packages/core/integration-tests/test/react-refresh.js

```diff
@@ -183,9 +186,12 @@ async function setup(entry) {
   });
 
   subscription = await b.watch();
-  let bundleEvent = await getNextBuild(b);
+  let bundleEvent: BuildEvent = await getNextBuild(b);
+  let bundleGraph;
+  if (bundleEvent.type === 'buildSuccess') {
```

Comment:
I'm sure this is probably just a hack to get flow to work, but can't this logic just be:

```js
assert.equal(bundleEvent.type, 'buildSuccess');
let bundleGraph = bundleEvent.bundleGraph;
```

Reply:
If flow doesn't refine the value, try `invariant` aliased to `assert`.